### DISCUSSION TITLE
OHBM 2024 Abstract: "fmriprep-next: Preprocessing as a fit-transform model"

### DISCUSSION THREAD

**Discussion by effigies** [nipreps/fmriprep#3130] [General]
Dear contributors,

We are planning on submitting an abstract to OHBM 2024. It is tentatively titled "fmriprep-next: Preprocessing as a fit-transform model", and there are the beginnings of a [draft](https://hackmd.io/1sKV_uG9S32KK2K4AnHgcg) on HackMD.

The topic of this abstract is the transition of fMRIPrep to a model where the  preprocessing is split into a minimal, compute-heavy "fit" stage that produces a small set of derivatives, and a "transform" stage where all other derivatives are deterministically generated.

Please feel free to make suggestions or comments. I aim to have a complete draft by ~~Monday, November 13~~ Tuesday, November 28. (I am simultaneously trying to release an alpha version of the work being described.)

If you feel you have contributed to this effort, please add your name, email and ORCID to the draft. Please also [register an account for abstract submission](https://ww6.aievolution.com/hbm2401/) regardless of whether you will be submitting an abstract yourself. By adding yourself to the system, I can add your information to the abstract exactly as you enter it. *If you are not registered, I will not add you manually.* I will be declaring that there are no conflicts of interest unless you specifically tell me to declare one on your behalf.

Submission closes on Friday, ~~November 17~~ December 1.

I would like to particularly invite @Lestropie, @36000, @feilong and @HippocampusGirl to add themselves to the author list for their contributions and discussions at the OHBM/Nipreps hacks in Glasgow and Montreal.

**Comment by Lestropie** [comment]
Very happy to see that there's progress being made on this. Sorry I haven't been able to contribute since our original discussion; I'm not even able to keep up with my own software at the moment, let alone others. But I'm still invested in the utilisation of these changes for the scientific application we discussed at the time. Hopefully over and above proposals for the abstract I can manage to chip in with some testing at least.

**Comment by effigies** [comment]
I have completed a draft that fits within the 4000 character limit. We have <100 characters to work with, so please bear that in mind for any suggestions.

The abstract is on [HackMD](https://hackmd.io/1sKV_uG9S32KK2K4AnHgcg), and I am attachign the [current preview](https://github.com/nipreps/fmriprep/files/13503226/preview.pdf) from the OHBM submission portal.

**Comment by effigies** [comment]
Here is a draft [poster_v1.pdf](https://github.com/user-attachments/files/15808182/poster_v1.pdf) for your review.

I feel like there are two target audiences for this poster.

1) "What are the use cases that make the workflow refactor make sense?"
2) "I know the use cases, how much better is it?"

I was planning on addressing both, but found that I had to choose between the manual intervention use case and benchmarks, in terms of space, and I think the manual intervention is more of a killer feature for people who don't have massive datasets to push to archives.

So I decided to create a QR code to a benchmarks page that I'll put together in the next couple of days, and that can be a part of the fMRIPrep docs going forward.

Happy to take any suggestions. If you have a grant you want credited for your contributions to this effort, LMK.

**Comment by oesteban** [comment]
Super cool Chris, thanks for this 👍 

Nothing jumped out when I skimmed it out. I like a lot using the two different colors for fit and transform consistently throughout.

**Comment by Lestropie** [comment]
Some suggestions: [poster_v1_RS.pdf](https://github.com/user-attachments/files/15813079/poster_v1_RS.pdf)

**Comment by poldrack** [comment]
LGTM overall - the only suggestion (optional) would be to credit scikit-learn for the fit/transform model.

**Comment by effigies** [comment]
Updated poster, incorporating @poldrack and @Lestropie's suggestions: 
[poster_v2.pdf](https://github.com/user-attachments/files/15841916/poster_v2.pdf)

**Reply by mgxd** [reply]
This looks great, Chris! Only one minor suggestion - the command line for the fit example could explicitly specify `minimal/` as the output directory, rather than `out/`
